### PR TITLE
MIGENG-416 setting LAZY to the relationship from AnalysisModel 

### PR DESCRIPTION
**Pull request by jonathanvila**
https://issues.redhat.com/browse/MIGENG-416
Changed to lazy the relationship between AnalysisModel and the list of VMWorkloadInventoryReportModel in order to reduce resources consumption

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/pojo/output/AnalysisModel.java [hunk 1]
```diff
@@ -32,11 +32,11 @@
     )
     private Long id;
 
-    @OneToOne(mappedBy = "analysis", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @OneToOne(mappedBy = "analysis", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
     @JsonIgnore
     private InitialSavingsEstimationReportModel initialSavingsEstimationReportModel;
 
-    @OneToMany(mappedBy = "analysis", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @OneToMany(mappedBy = "analysis", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
```
Comment (carlosthe19916) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/AnalysisModel.java:39]: 1. After we change from EAGER to LAZY the deprecated method `addWorkloadInventoryReportModel` is not going to work anymore, which is expected and ok. I suggest to delete this deprecated method in this PR. https://github.com/project-xavier/xavier-integration/blob/292c2494f912d7289119b4a8af365265b117d057/src/main/java/org/jboss/xavier/integrations/jpa/service/AnalysisService.java#L69

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/analytics/pojo/output/AnalysisModel.java:39]: +1, if fact I dont understand why we did set it to deprecated and no other method in the code was using it.....

**Review by carlosthe19916 — APPROVED** [review]
This PR looks good to me.

**Review by mrizzi — COMMENTED** [review]
I ran a test in CI comparing `master` branch performance with the one of this PR and there a nice improvement :+1: 
@mperezco was able to see his last report (`1033`) with this PR as well (previously he just received "Failed to load reports" notifications)

**Pull request merged**